### ISSUE TITLE
rpc error when changing existing VMs network_interface

### ISSUE THREAD

**Issue by ol-metaswitch** [terraform-providers/terraform-provider-vsphere#822]
### Terraform Version

0.12.4

### vSphere Provider Version

1.12.0

### Affected Resource(s)

- imported `vsphere_virtual_machine`

### Terraform Configuration Files

```hcl
provider "vsphere" {
  allow_unverified_ssl = true
  client_debug = true
}

data "vsphere_datacenter" "dc" {
  name = "dc_name"
}

data "vsphere_datastore_cluster" "datastore_cluster" {
  name          = "datastore_cluster_name"
  datacenter_id = data.vsphere_datacenter.dc.id
}

data "vsphere_network" "network" {
  name          = "network_name"
  datacenter_id = data.vsphere_datacenter.dc.id
}

data "vsphere_virtual_machine" "vm_template" {
  name          = "gitlab-runner-coreos-2135.5.0"
  datacenter_id = data.vsphere_datacenter.dc.id
}

resource "vsphere_virtual_machine" "devel-runners" {
  count = 2
  name                 = "machine${count.index + 1}"
  resource_pool_id     = "resgroup-50029"
  datastore_cluster_id = data.vsphere_datastore_cluster.datastore_cluster.id
  enable_logging = true
  folder = "devel-runner folder/"
  lifecycle {
    ignore_changes = ["custom_attributes", "imported", "clone"]
  }
  num_cpus = 2
  cpu_share_level = "custom"
  memory   = 12288
  memory_share_level = "custom"
  guest_id = data.vsphere_virtual_machine.vm_template.guest_id

  network_interface {
    network_id   = data.vsphere_network.network.id
    adapter_type = data.vsphere_virtual_machine.vm_template.network_interface_types[0]
  }

  disk {
    label = "disk0"
    size  = data.vsphere_virtual_machine.vm_template.disks[0].size
    keep_on_remove = true
  }

  clone {
    template_uuid = data.vsphere_virtual_machine.vm_template.id
  }

  vapp {
    properties = {
      "guestinfo.coreos.config.url" = "https://config.url/"
      "guestinfo.hostname"          = "machine${count.index + 1}"
    }
  }
}
```

### Panic Output

https://gist.github.com/ol-metaswitch/44ac21c8e372c06005f2451ab15bc965

### Expected Behavior

The changes to the network interface in the `terraform apply` to be made to the VM are made without error.

### Actual Behavior

Upon accepting the changes to the network interface made by the `terraform apply` command Terraform crashed.

### Steps to Reproduce

Please list the steps required to reproduce the issue, for example:

1. Import existing VM with a different `network_interface["network_id"]` to the one stated into terraform state
2. `terraform apply`
3. Accept changes indicated from the apply
4. Terraform crashes

**Comment by ol-metaswitch** [comment]
After some further testing I was able to narrow this problem down using the `ignore_changes` attribute. It seems as though terraform is able to make all the changes except for the `network_interface["network_id"]` so that should narrow things down quite a lot. I'll edit the original description to reflect this new knowledge.

**Comment by larhauga** [comment]
@ol-metaswitch would you be able to test the PR referenced #840? I had the same issue when cloning a template and trying to configure another network id.

**Comment by ol-metaswitch** [comment]
Yep, I'll test it out tomorrow and comment on here with what I find.

**Comment by ol-metaswitch** [comment]
I've tried building the provider as described in the README but I get the following error:
```
make build
==> Checking that code complies with gofmt requirements...
gofmt needs running on the following files:
./vsphere/distributed_virtual_switch_structure.go
./vsphere/resource_vsphere_compute_cluster_vm_dependency_rule.go
./vsphere/resource_vsphere_compute_cluster_vm_host_rule.go
./vsphere/resource_vsphere_resource_pool.go
You can use the command: `make fmt` to reformat code.
make: *** [fmtcheck] Error 1
```

**Comment by ol-metaswitch** [comment]
Should I just run the `make fmt` command?

**Comment by ol-metaswitch** [comment]
I managed to figure it out after running some format commands and putting the binaries in the correct places. I still get a crash though the crash log is available here:

https://gist.github.com/ol-metaswitch/18224cf376c6e457843de93d215cbaa5

**Comment by larhauga** [comment]
Would you mind trying out the following: https://gist.github.com/larhauga/88caf283d0fd6f9c3d1c848cad1d8c19
I would like to compare it to the output I am getting.
I am not sure but there could be an issue in the govmomi package that are causing this.
Also make sure that you have cloned out my repo+branch (cant se any versions in the output) ;)

Thanks!

**Comment by wkleinhenz** [comment]
im having similar issues and built youre fix @larhauga but am still having the crash so im attaching the logs after making the changes in your gist. i can add my tf file if needed.
[terraform crash log.txt](https://github.com/terraform-providers/terraform-provider-vsphere/files/3665745/terraform.crash.log.txt)

versions
terraform 0.12.9
vsphere provider 1.12

**Comment by koikonom** [comment]
Hi @ol-metaswitch, we have just merged PR #840 that I believe will address this issue. It is planned for the next provider release, so keep an eye for the new version.

**Comment by ol-metaswitch** [comment]
Hi @koikonom , sorry about the delayed reply I've been away/busy since you've been working hard on this. I might be able to test out the changes you've made in the next few days but if I don't find the time it may not be until 3 weeks time.

Again, thanks for your help and I'll comment back on here when I get a chance to test any changes out.

**Comment by bill-rich** [comment]
Looks like this is hopefully resolved with #840. Please let us know if it is still a problem and we'll get the issue reopened.
